security/pkg/registry/kube/service.go

```diff
@@ -23,32 +24,30 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pilot/platform/kube"
+	"istio.io/istio/security/pkg/registry"
 )
 
-// ServiceController monitors the service definition changes in a namespace.
-// Callback functions are called whenever there is a new service, a service deleted,
-// or a service updated.
+// ServiceController monitors the service definition changes in a namespace. If a
+// new service is added with "alpha.istio.io/kubernetes-serviceaccounts" annotation
```

Comment:
Note there is a legacy issue in my previous PR, ideally we want to use CanonicalServiceaccount... as mentioned here for non-k8s service account. https://github.com/istio/istio/blob/a1ba1fbf9cdeac6bd51898732f787948dcc9d160/pilot/platform/kube/conversion.go#L43

But as of now, let's just make it work and adjust later.

Reply:
Cool. I'll add a TODO